pymc_marketing/mmm/models/components/base.py

```diff
@@ -0,0 +1,155 @@
+import warnings
+from inspect import signature
+from typing import Any
+
+from pytensor import tensor as pt
+
+from pymc_marketing.mmm.utils import _get_distribution
+
+
+class ParameterPriorException(Exception):
+    """Error when the functions and specified priors don't match up."""
+
+    def __init__(self, priors: set[str], parameters: set[str]) -> None:
+        self.priors = priors
+        self.parameters = parameters
+
+        msg = "The function parameters and priors don't line up."
+
+        if self.priors:
+            msg = f"{msg} Missing prior: {self.priors}."
+
+        if self.parameters:
+            msg = f"{msg} Missing parameter: {self.parameters}."
+
+        super().__init__(msg)
+
+
+class Transformation:
+    """Base class for adstock and saturation functions.
+
+    The subclasses will need to implement the following attributes:
+
+    - function: The function that will be applied to the data.
+    - prefix: The prefix for the variables that will be created.
+    - default_priors: The default priors for the parameters of the function.
+
+    In order to make a new saturation or adstock function, this subclass would
```

Comment:
Do you mean "this ~~sub~~class would not be used"?

Reply:
```suggestion
    In order to make a new saturation or adstock function, this class would not be used but rather `SaturationTransformation` and `AdstockTransformation`
```